jobbergate-agent/jobbergate_agent/jobbergate/update.py

```diff
@@ -0,0 +1,95 @@
+import json
+from typing import List
+
+from loguru import logger
+
+from jobbergate_agent.clients.cluster_api import backend_client as jobbergate_api_client
+from jobbergate_agent.clients.slurmrestd import backend_client as slurmrestd_client
+
+# from jobbergate_agent.jobbergate.api import update_status
+from jobbergate_agent.jobbergate.schemas import ActiveJobSubmission, SlurmJobData
+from jobbergate_agent.utils.exception import JobbergateApiError, SlurmrestdError
+from jobbergate_agent.utils.logging import log_error
+
+
+async def fetch_job_data(slurm_job_id: int) -> SlurmJobData:
+    logger.debug(f"Fetching slurm job status for slurm job {slurm_job_id}")
+
+    response = await slurmrestd_client.get(f"/job/{slurm_job_id}")
+
+    with SlurmrestdError.handle_errors("Failed to fetch job state from slurm", do_except=log_error):
+        response.raise_for_status()
+        data = response.json()
+
+    jobs = data["jobs"]
+    SlurmrestdError.require_condition(
```

Comment:
I think this would be a good case to mark the jobs as `unknow` status. Because if slurm doesn't know about them and we keep the status as it was, they will be continues pulled by the agent just to raise this error again.

Reply:
Updated in commit e2e8554.